### PR TITLE
Tyler_Chittum-Signoff-RRSA

### PR DESCRIPTION
**Pull request by branchworm**
Requesting signoff for RRSA.

### CODE REVIEW

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+### Constraints
+
+**Measurement accuracy:** it is imperative that the device record the altitude of the rider with a high degree of precision. If the simulated resistance is to be within 10 % of the real-world resistance as previously constrained, the combined inaccuracy of the pressure sensor, the speed sensor, and the resistance system must be below 10 %. Let the error allowance for the determination of the energy expended during exercise be 1 %. For a bike and rider of 91 kg, an altitude gain of 4.67 m results in an energy expenditure of 1 kCal since E = mgh. Accordingly, the required precision to achieve +/- 1 % accuracy at a granularity of 1 kCal is +/- 4.67 cm. The speed must also be sampled at a rate which allows reasonably fast response time. Since the use of single-magnet speed sensing is already proven as common practice for bike computers, and all potential energy calculations can be performed with only the measured altitude over time, the main constraint for speed sensing is one of durability. The reed switch used must nominally tolerate at least 1000 miles of operation.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md]: This is a bit uncertain as to what is meant by reasonably fast response time etc. So, this is not quite precise. Though it sounds like you are on the right track.

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+![RRSA Schematic](https://user-images.githubusercontent.com/118228609/203230366-a49b449b-943c-4f00-ad5b-cdeb976bec66.png)
+Figure 1: a schematic for the assembly. Not pictured: the camera, which is to be attached to the camera port on the Raspberry Pi Zero 2W.
+
+![PIC03042](https://user-images.githubusercontent.com/118228609/203231527-3cd93c0c-69ab-4919-86b1-4a5de69bc6ae.JPG)
+Figure 2: a drawing illustrating the placement of modules within the project box.
+
+![PIC03041](https://user-images.githubusercontent.com/118228609/203231351-a6e881cc-bc34-441a-97ff-cbec0ddabef7.JPG)
+Figure 3: a drawing illustrating the front and back panels. The first item (from left to right) on the front panel is a panel-mount USB-C socket. The second is the power switch, followed by the start/stop switch. The hole in the back panel is for the camera.
+
+### Analysis:
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md:55]: For the sensing appropriateness and battery life, more numeric analysis is needed.

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+**Battery life:** the device must be able to continuously operate for at least 3 hours, and must provide the user with a visual indication of battery charge status. 
+
+**Recording capacity:** the device must be able to capture 3 hours of continuous footage and measurements. Additionally, the storage media must possess the bandwidth to record the video as it is being taken. 
+
+**Usability:** the device must provide an easily used interface, which is also unobtrusive and does not distract the rider. 
+
+**Durability:** the device must be shielded from outside weather conditions, and all component boards must be attached firmly enough to prevent them from being knocked around during riding.
+
+### Schematic:
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md:35]: This seems to be a 3d print situation for the case but there are no 3d models for schematics. A mounting system and storage schematic is needed.

**Review by JesseTNRoberts — COMMENTED** [review]
Very good first attempt!

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md (hunk 2 above)
Comment (branchworm) [reply Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md:55]: Battery life analysis completed.

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+The utilized pressure sensor is specified to provide a measurement precision of +/- 2 cm, so it satisfies the prior specification. In addition, it provides on-board measurement averaging and low-pass filtering to reduce noise from the measurement.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md]: Across what range?

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+The utilized pressure sensor is specified to provide a measurement precision of +/- 2 cm, so it satisfies the prior specification. In addition, it provides on-board measurement averaging and low-pass filtering to reduce noise from the measurement. 
+
+Hammlin specifies that their reed switches can last over 200 million actuations under ideal conditions [6]. The rated current for these reed switches is 0.5 A, which is well above what would be required to provide a signal voltage to the computer. Assuming a 200-million-actuation life and a 26" bike tire, the device would last for 257,732 miles, satisfying the aforementioned specification.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md]: What is the read switch for? I don't recall a constraint for this.

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+The utilized pressure sensor is specified to provide a measurement precision of +/- 2 cm, so it satisfies the prior specification. In addition, it provides on-board measurement averaging and low-pass filtering to reduce noise from the measurement. 
+
+Hammlin specifies that their reed switches can last over 200 million actuations under ideal conditions [6]. The rated current for these reed switches is 0.5 A, which is well above what would be required to provide a signal voltage to the computer. Assuming a 200-million-actuation life and a 26" bike tire, the device would last for 257,732 miles, satisfying the aforementioned specification. 
+
+The selected camera is capable of recording video in HD, 60 fps (or below) and thus satisfies the previously mentioned specification.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md]: Is this the complete set of constraints for the camera? What about format? How does the system sync the data to the footage?

Comment (JesseTNRoberts) [reply Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md]: How does the video get to the PI?

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+The interface is a simple, unobtrusive, two button one with a light integrated into each button. One button turns the unit on or off, and the light for this button lights up when the unit is powered. The other button acts to activate or stop the recording. When this button is pressed, recording begins, and the light lights up. When the button is pressed again, recording stops, and the light turns off when the Raspberry Pi enters low power mode. This light also flashes when data transfer is in progress.
+
+By gluing the boards down inside the case, they can be solidly anchored to the inside of the box. In addition, all holes, except what is necessary to allow airflow to the pressure sensor may be coated with glue to waterproof them.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md]: Gluing? I think that is a bad idea.

Comment (JesseTNRoberts) [reply Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md]: How will this be worked on? How will you replace components?

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+To meet the mountability constraint, the housing for this device is to be secured to the bike using an already-existing universal bike mounting bracket designed for mounting cell phones and similar objects using a combination of adhesive coupling to the device housing, a mechanical twist-locking mechanism, and a screw-tensioned ring with rubber adaptors to allow use on a multitude of handlebar sizes. The angle of the device is adjustable by the user, ensuring the ability to attain an ideal camera angle. The twist-lock mechanism provided by this bracket provides a means to remove the device very rapidly, and without the use of any tools.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - RRSA System.md]: No schematic showing this and the housing being attached.

**Review by JesseTNRoberts — COMMENTED** [review]
Hand drawn schematics are a good place to start. However, if you intend to print this, you will need a 3d model. The 3d model should be completed now to show how the components will fit and how it will mount. Otherwise, it is still a guess.

**Comment by branchworm** [comment]
> Hand drawn schematics are a good place to start. However, if you intend to print this, you will need a 3d model. The 3d model should be completed now to show how the components will fit and how it will mount. Otherwise, it is still a guess.

I don't intend to print it; rather, I intend to purchase a pre-made project box of the dimensions shown in the diagram, which is included in the BOM.

**Review by JesseTNRoberts — APPROVED** [review]

**Pull request merged**